addon/components/ember-wormhole.js

```diff
@@ -1,25 +1,90 @@
 import Ember from 'ember';
+import layout from '../templates/components/ember-wormhole';
 
 var computed = Ember.computed;
 var observer = Ember.observer;
 var run = Ember.run;
 
+function getActiveElement() {
+  if (typeof document === 'undefined') {
+    return null;
+  } else {
+    return document.activeElement;
+  }
+}
+
+// used to find an element by id
+function childNodesOfElement(element) {
+  let children = [];
+  let child = element.firstChild;
+  while (child) {
+    children.push(child);
+    child = child.nextSibling;
+  }
+  return children;
+}
+
+// implement our own getElementById since it doesn't exist in SimpleDOM
+function findElementById(doc, id) {
+  let nodes = childNodesOfElement(doc);
+  let node;
+
+  while (nodes.length) {
+    node = nodes.shift();
+
+    if (node.getAttribute && node.getAttribute('id') === id) {
+      return node;
+    }
+
+    nodes = childNodesOfElement(node).concat(nodes);
+  }
+}
+
 export default Ember.Component.extend({
+  layout,
   to: computed.alias('destinationElementId'),
   destinationElementId: null,
   destinationElement: computed('destinationElementId', 'renderInPlace', function() {
```

Comment:
I know this is refactoring something already here, but this should not be a CP, we should just have a findDestination when we need to update the destination.  The component hooks will be invoked if the input has changed.  This isn't a property that is actually bound to anything outside of the component itself.  If the hooks are invoked for didUpdate, you can compare if your head's parentNode matches the new location.